### DISCUSSION TITLE
What is the proper bounded channel size in server side streaming?

### DISCUSSION THREAD

**Discussion by YjyJeff** [hyperium/tonic#1129] [Q&A] [answered]
Hi, I want to implement a server-side streaming application and am confused about the channel size. I am using `tokio_stream::wrappers::ReceiverStream` as the return type. I found that the server will send all of the responses no matter whether the client is polled or not, with the bounded channel. 

For example, when I set the channel size to `1` and does not poll the receiver on the client. The server can send `100`  responses to the client. So what is the usage of the bounded channel here if I can send all of the responses?  Does the server really sent the responses to the client  or just buffer the response in memory and sent it until the client is polled on the receiver?

Thanks in advance, sincerely

**Comment by LucioFranco** [comment, marked as answer]
In this context, what do you mean client polled? The server will attempt to fill the tcp buffer eagerly and backpressure is done via tcp/h2 windowing. Not sure if that answers your question but the client doesn't need to poll to have data sent on the wire.

**Reply by YjyJeff** [reply]
This is exactly what I want, thanks~
